zerver/models.py

```diff
@@ -315,11 +319,19 @@ def get_realm_by_email_domain(email):
     if settings.REALMS_HAVE_SUBDOMAINS:
         raise GetRealmByDomainException(
             "Cannot get realm from email domain when settings.REALMS_HAVE_SUBDOMAINS = True")
-    try:
-        alias = RealmAlias.objects.select_related('realm').get(domain = email_to_domain(email))
+    domain = email_to_domain(email)
+    query = RealmAlias.objects.select_related('realm')
+    alias = query.filter(domain=domain).first()
+    if alias is not None:
         return alias.realm
-    except RealmAlias.DoesNotExist:
-        return None
+    else:
```

Comment:
we should definitely clean this up; since we have pretty complete tests for these messy functions, I'm content to merge this and clean up afterwards, but we should definitely make them nice and readable as a follow-up.